### PR TITLE
Add GitHub Actions to publish dash

### PR DESCRIPTION
**Pull request by thisisnic**

### CODE REVIEW

**Comment by thisisnic** [comment]
OK, the latest is that I've currently got this mostly rendering, but the calls to the itables `show()` function doesn't work now I removed the jupyter setting (which I had to do to get the R stuff to work) so I commented them out.  I will try later swapping out those functions for an R equivalent to see if that works (or happy for a Py alternative but I don't know any!

**Comment by AlenkaF** [comment]
I will check out for Python alternatives as the data is already all in Python (for the GitHub part).

File: requirements.txt
Comment (AlenkaF) [inline requirements.txt:1]: I think most of the packages are installed together with `Jupyter` installation so could be omitted here. This was the list of installed dependencies when I installed `Jupyter` locally:

<details>
 <summary> Open for full list of packages ...</summary>

```
Package                   Version
------------------------- --------------
anyio                     4.4.0
appnope                   0.1.4
argon2-cffi               23.1.0
argon2-cffi-bindings      21.2.0
arrow                     1.3.0
asttokens                 2.4.1
async-lru                 2.0.4
attrs                     23.2.0
Babel                     2.15.0
beautifulsoup4            4.12.3
bleach                    6.1.0
certifi                   2024.6.2
cffi                      1.16.0
charset-normalizer        3.3.2
comm                      0.2.2
debugpy                   1.8.1
decorator                 5.1.1
defusedxml                0.7.1
executing                 2.0.1
fastjsonschema            2.20.0
fqdn                      1.5.1
h11                       0.14.0
httpcore                  1.0.5
httpx                     0.27.0
idna                      3.7
ipykernel                 6.29.4
ipython                   8.25.0
ipywidgets                8.1.3
isoduration               20.11.0
jedi                      0.19.1
Jinja2                    3.1.4
json5                     0.9.25
jsonpointer               3.0.0
jsonschema                4.22.0
jsonschema-specifications 2023.12.1
jupyter                   1.0.0
jupyter_client            8.6.2
jupyter-console           6.6.3
jupyter_core              5.7.2
jupyter-events            0.10.0
jupyter-lsp               2.2.5
jupyter_server            2.14.1
jupyter_server_terminals  0.5.3
jupyterlab                4.2.2
jupyterlab_pygments       0.3.0
jupyterlab_server         2.27.2
jupyterlab_widgets        3.0.11
MarkupSafe                2.1.5
matplotlib-inline         0.1.7
mistune                   3.0.2
nbclient                  0.10.0
nbconvert                 7.16.4
nbformat                  5.10.4
nest-asyncio              1.6.0
notebook                  7.2.1
notebook_shim             0.2.4
overrides                 7.7.0
packaging                 24.1
pandocfilters             1.5.1
parso                     0.8.4
pexpect                   4.9.0
pip                       24.0
platformdirs              4.2.2
prometheus_client         0.20.0
prompt_toolkit            3.0.47
psutil                    6.0.0
ptyprocess                0.7.0
pure-eval                 0.2.2
pycparser                 2.22
Pygments                  2.18.0
python-dateutil           2.9.0.post0
python-json-logger        2.0.7
PyYAML                    6.0.1
pyzmq                     26.0.3
qtconsole                 5.5.2
QtPy                      2.4.1
referencing               0.35.1
requests                  2.32.3
rfc3339-validator         0.1.4
rfc3986-validator         0.1.1
rpds-py                   0.18.1
Send2Trash                1.8.3
setuptools                70.0.0
six                       1.16.0
sniffio                   1.3.1
soupsieve                 2.5
stack-data                0.6.3
terminado                 0.18.1
tinycss2                  1.3.0
tornado                   6.4.1
traitlets                 5.14.3
types-python-dateutil     2.9.0.20240316
uri-template              1.3.0
urllib3                   2.2.2
wcwidth                   0.2.13
webcolors                 24.6.0
webencodings              0.5.1
websocket-client          1.8.0
widgetsnbextension        4.0.11
```

</details>

I would also unpin the package versions.

**Comment by AlenkaF** [comment]
> OK, the latest is that I've currently got this mostly rendering, but the calls to the itables `show()` function doesn't work now I removed the jupyter setting (which I had to do to get the R stuff to work) so I commented them out. I will try later swapping out those functions for an R equivalent to see if that works (or happy for a Py alternative but I don't know any!

If I am correct you are referring to `-jupyter: python3` in the top of the Quarto page.
I have removed it and it works for me locally. Not sure what could be the issue but maybe there is an empty line missing after newly added "update" section. I am installing R at the moment to test it out.

**Comment by AlenkaF** [comment]
Ok, got there! :D Using R and Python with `Knitr` and `reticulate` renders the quarto file but instead of tables `<IPython.core.display.HTML object>` is shown. Will try to fix it.

**Comment by thisisnic** [comment]
> I will check out for Python alternatives as the data is already all in Python (for the GitHub part).

I think quarto can use reticulate under the hood to easily pass the data between R and Python, so we could use some R ones too- might be nice to demo how easy it is to swap between? 

Or, stick with the Python ones for that and I'll do another output with an R library but Python data.

File: requirements.txt
Comment (thisisnic) [reply requirements.txt:1]: Thanks, let's discuss later as I'm pretty clueless about all of this so need your help!

**Comment by AlenkaF** [comment]
> I think quarto can use reticulate under the hood to easily pass the data between R and Python, so we could use some R ones too- might be nice to demo how easy it is to swap between?

Yeah, that would be great actually.

The interactive tables created with `itables` produce `<IPython.core.display.HTML object>` which need `Jupyter` engine as far as I understand. One combo to make Quatro work with both R and Python is `Knitr` and `reticulate`, the other is `Jupyter` and `rpy2`. Unfortunately I haven't been able to make any of those two options work yet :D So the solution you mention might be the way to go.

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+types-python-dateutil==2.9.0.20240316
+typing_extensions==4.12.2
+tzdata==2024.1
+uri-template==1.3.0
+urllib3==2.2.2
+wcwidth==0.2.13
+webcolors==24.6.0
+webencodings==0.5.1
+websocket-client==1.8.0
+widgetsnbextension==4.0.11
```
Comment (AlenkaF) [inline requirements.txt:111]:
```suggestion
pandas
altair
```

Can we try with this - i think it should work.

File: index.qmd [hunk 2]
```diff
@@ -17,6 +16,32 @@ Seek inspiration from https://quarto.org/docs/dashboards/examples/
...
+  DT::datatable(
+    x,
+    escape = FALSE,
+    extensions = 'Buttons',
+    options = list(
+      dom = 'Bfrtip',
+      buttons = c('copy', 'csv', 'excel')
+    )
+  )
+}
```
Comment (AlenkaF) [inline index.qmd:41]: Love it! 🎉

**Review by AlenkaF — COMMENTED** [review]
This looks great! (will try it out locally after I get a cup of coffee 😄 )

**Comment by AlenkaF** [comment]
> (will try it out locally after I get a cup of coffee 😄 )

Yeah, don't have to! 🤣 
https://thisisnic.github.io/arrowdash/# ❤️

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+pandas
+altair
```
Comment (thisisnic) [inline requirements.txt:2]:
```suggestion
altair
requests
```

**Comment by thisisnic** [comment]
@AlenkaF I couldn't get it working with the trimmed-down requirements, so I reverted those changes for now, just for the sake of closing this PR, but let's revisit this!

**Pull request merged**